src/main/java/uk/gov/justice/digital/delius/transformers/InstitutionalReportTransformer.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.delius.transformers;
+
+import org.springframework.stereotype.Component;
+import uk.gov.justice.digital.delius.data.api.Conviction;
+import uk.gov.justice.digital.delius.data.api.InstitutionalReport;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Custody;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Disposal;
+
+import java.util.Optional;
+
+import static uk.gov.justice.digital.delius.transformers.TypesTransformer.convertToBoolean;
+import static uk.gov.justice.digital.delius.transformers.TypesTransformer.zeroOneToBoolean;
+
+@Component
+public class InstitutionalReportTransformer {
+
+    public InstitutionalReportTransformer() {
+    }
+
+    public InstitutionalReport institutionalReportOf(uk.gov.justice.digital.delius.jpa.standard.entity.InstitutionalReport report) {
+        return InstitutionalReport.builder()
+            .institutionalReportId(report.getInstitutionalReportId())
+            .offenderId(report.getOffenderId())
+            .conviction(convictionOf(report))
+            .build();
+    }
+
+    private Conviction convictionOf(uk.gov.justice.digital.delius.jpa.standard.entity.InstitutionalReport report) {
+
+        return Optional.ofNullable(report.getCustody())
+            .filter(custody -> !convertToBoolean(custody.getSoftDeleted()))
```

Comment:
Minor observation; I bet this can't happen in the system; e.g we have a non-deleted report associated with a deleted custody/disposal record. But the code does no harm so no need to change

Reply:
Hopefully it would never happen. This code protects us if it does. BCL code or DB triggers should keep all the soft deleted flags in sync across connected records